src/main.h

```diff
@@ -632,7 +636,7 @@ uint64_t CalculateCurrentUsage();
 CMutableTransaction CreateNewContextualCMutableTransaction(
     const Consensus::Params& consensusParams,
     int nHeight,
-    bool requireSprout = false);
+    bool requireV4);
```

Comment:
Hmm, I'm not sure this is sufficient. We have several situations here that a transaction builder could be in:

- Tx involving Sprout: we need to ignore the new flag and require v4.
- Tx involving Orchard: we need to ignore the new flag and require v5.
- Tx involving neither Sprout nor Orchard: we respect the new flag.

Main point is that the user's desire to spend Sprout or Orchard funds is something that needs to either override the new flag, or (in the case of Orchard) we need to take the flag into account when deciding whether to spend Orchard funds.

Reply:
Previously, the `requireSprout` flag was respected unconditionally. I think that it's correct to continue to do this, and to ensure we pass the correct value based on that flag at the new call site. Getting rid of the default ensures we have to always make an explicit choice.

Reply:
We'll need to change this to `int preferredTxVersion` anyway, I think.

Reply:
At this stage we should keep it a boolean value here, because it's a hard constraint - we can switch it later when we have more than two versions to worry about.